ksiren/src/main/java/com/brightspace/ksiren/DefinedEntity.kt

```diff
@@ -15,9 +15,38 @@ package com.brightspace.ksiren
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-abstract class DefinedEntity(entity: Entity, signature: KSirenEntitySignature) {
+abstract class DefinedEntity(val entity: Entity, signature: KSirenEntitySignature) {
+
+	val subEntities: MutableList<Entity> = entity.entities.toMutableList()
```

Comment:
Do we want to expose a mutable list here? We can either keep it private and force access to the subentities through `getSubEntitiesOfType`, which is good because it will return instances of `DefinedEntity` or we can expose it as an immutable list of "raw" entities. I'd favour the first approach myself as we can always get the raw entities from the embedded `entity` property in this class.

Reply:
Right, right; I agree with the first approach too, I just didn't make the mental leap from `Entity` to `DefinedEntity`. I'll make another commit.